### PR TITLE
Free some in-memory usage when doing HPOS-related batch processing

### PR DESCRIPTION
**Pull request by gedex**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Running `wc hpos cleanup all` on a store with 80GB+ wp_postmeta will likely hitting OOM, even with big memory limit. This PR adds in-memory cleanup from https://github.com/woocommerce/woocommerce/pull/44670 with the addition of clearing wpdb query log and moving it into a new method `free_in_memory_usage()` so that it can be used by other batch-processing CLI commands.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add debug line to note peak memory usage

   ```php
   <?php
   add_action(
       'shutdown',
       function() {
    	   printf( 'Peak memory usage: %s', size_format( memory_get_peak_usage(), 2 ) );
       }
   );
   ```

2. Make sure HPOS enabled.
3. Generate the orders, `wp wc generate orders 1000000`. In my case, peaked memory usage wasn't noticeable with small number of orders.
4. Sync to CPT, `wp wc hpos sync`.
5. Run `wp wc hpos cleanup all` .

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Review by jorgeatorres — APPROVED** [review]
LGTM. Thanks @gedex! 🙇

**Comment by gedex** [comment]
I tried re-rerun failed but the core e2e tests always failed. It doesn't seem relate to the change in the PRs, so it's likely:

* hitting the limit of free limit as the error suggests: `You have exceeded the number of executions available on the free plan. Please upgrade to a paid plan to continue processing test executions.`. or
* flaky e2e tests

**Comment by vedanshujain** [comment]
Yeah looks like a 502 error code. All tests do pass, just that the posting result thing does not work. Let me try again, otherwise will take a more detailed look.

**Comment by jorgeatorres** [comment]
I believe those errors were recently fixed in `trunk`. I have rebased. Let's hope that addresses the CI failures.
